app/forms/candidate_interface/course_selection/which_course_are_you_applying_to_step.rb

```diff
@@ -50,11 +50,14 @@ def next_step
           :course_study_mode
         elsif multiple_sites?
           :course_site
+        else
+          :course_review
         end
       end
 
       def next_edit_step_path(next_step_klass)
-        return next_step_path(next_step_klass) unless next_step_klass.new.next_step
+        classes_without_edit = [DuplicateCourseSelectionStep, FullCourseSelectionStep, ClosedCourseSelectionStep]
+        return next_step_path(next_step_klass) if classes_without_edit.include?(next_step_klass)
```

Comment:
This is a bug fix.

We get silent exception.

When we try to instantiate the next step and call `next_step` on that (to know if there is a next step) we get errors about the params required to create the next - next step.

This was originally designed to avoid having to list all the steps we know do not have a next step.

We now revert to doing that out of necessity

Reply:
Ideally "terminal" or "editable" steps should have a property so we can identify them.

```ruby
class ThingStep < DfE::Wizard::Step

  def self.terminal
    true
  end

...

return next_step_path(next_step_klass) unless next_step_klass.terminal?

```

@tomadams-dfe

Reply:
I try to read and read and I don't understand to be fair. 🤔 

Let's huddle in the afternoon?